folium/features.py

```diff
@@ -1973,3 +1985,65 @@ def __init__(
             out.setdefault(cm(color), []).append([[lat1, lng1], [lat2, lng2]])
         for key, val in out.items():
             self.add_child(PolyLine(val, color=key, weight=weight, opacity=opacity))
+
+
+class Control(MacroElement):
```

Comment:
Good point and the answer is I don't know yet. 

I noted that the `CustomControl` is like a `Control` that is linked to a layer, something I did not realize at first. In pure `Leaflet` this is something of an anomaly, as both `Control`s and `Layer`s are contained in a `Map` and there is no structural association between `Control` and `Layer`. Typically it is the other way round and a `Control` determines the visibility of the `Layer`.

My guess is it would be fine as you are implementing it now. It would probably be the same as other `Controls` that are associated with a set of `Layers`.

Reply:
> I noted that the CustomControl is like a Control that is linked to a layer, something I did not realize at first. In pure Leaflet this is something of an anomaly, as both Controls and Layers are contained in a Map and there is no structural association between Control and Layer. Typically it is the other way round and a Control determines the visibility of the Layer.

That's interesting... So I guess the custom control is a bit of a weird one, and it's okay not to consider it here.